tests/FunctionalTestCase.php

```diff
@@ -385,6 +395,16 @@ protected function isReplicaSet()
         return $this->getPrimaryServer()->getType() == Server::TYPE_RS_PRIMARY;
     }
 
+    /**
+     * Return whether serverless (i.e. proxy as mongos) is being utilized.
+     */
+    protected static function isServerless(): bool
+    {
+        $isServerless = getenv('MONGODB_IS_SERVERLESS');
+
+        return $isServerless !== false ? filter_var($isServerless, FILTER_VALIDATE_BOOLEAN) : false;
```

Comment:
Do we actually need all of the possible values detected by [`FILTER_VALIDATE_BOOLEAN`](https://www.php.net/manual/en/filter.filters.validate.php)? I don't see the harm in doing this, but just curious if it makes sense to do so vs. being very strict about only supporting particular values that the Evergreen config might specify.

Reply:
The decision was mostly personal preference, since I can't remember which tools take which formats in their bool environment variables. That's why I opted for something that is as lenient as possible, especially when `on`, `yes`, and `1` all make sense for the behaviour.